lib/src/source/hosted.dart

```diff
@@ -48,6 +48,15 @@ Uri validateAndNormalizeHostedUrl(String hostedUrl) {
   Uri u;
   try {
     u = Uri.parse(hostedUrl);
+
+    // pub.dev and pub.dartlang.org are identical.
+    //
+    // We rewrite here to avoid caching both, and to avoid having different
+    // credentials for these two.
+    if (u == Uri.parse('https://pub.dev')) {
+      log.message('Using https://pub.dartlang.org instead of https://pub.dev.');
+      u = Uri.parse('https://pub.dartlang.org');
+    }
```

Comment:
You might want to do this after we've done:
```dart
  u = u.normalizePath();
  // If we have a path of only `/`
  if (u.path == '/') {
    u = u.replace(path: '');
  }
  // If there is a path, and it doesn't end in a slash we normalize to slash
  if (u.path.isNotEmpty && !u.path.endsWith('/')) {
    u = u.replace(path: u.path + '/');
  }
```

So right before return.. Otherwise, we might have`https://pub.dev/` getting normalized to `https://pub.dev`, but `https://pub.dev` gets normalized to `https://pub.dartlang.org` :rofl:

Reply:
Good catch, done!